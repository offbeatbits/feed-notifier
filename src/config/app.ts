export const enableLogging = process.env.LOGGING_ENABLE! === 'true' || false;
export const enableVerboseLogging = process.env.LOGGING_VERBOSE_ENABLE! === 'true' || false;
