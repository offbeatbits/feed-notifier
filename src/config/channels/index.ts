import { mastodon } from './mastodon';
import { twitter } from './twitter';

export const channels = {
  mastodon,
  twitter,
};
