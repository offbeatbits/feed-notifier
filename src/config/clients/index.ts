import { mastodon } from './mastodon';
import { twitter } from './twitter';

export const clients = {
  mastodon,
  twitter,
};
