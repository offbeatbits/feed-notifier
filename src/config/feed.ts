export const feedUrl = process.env.FN_FEED_URL!;
