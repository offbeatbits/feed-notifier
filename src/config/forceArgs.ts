export const forceArgs = [
  '-f',
  '-F',
  '-force',
  '-FORCE',
  '--force',
  '--FORCE',
  '--f',
  '--F',
  'force',
  'FORCE',
];
