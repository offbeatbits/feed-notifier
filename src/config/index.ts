export * as app from './app';
export * from './serviceMappings';
export * from './feed';
export * from './forceArgs';
export * from './initArgs';
export * as channels from './channels';
export * as clients from './clients';
