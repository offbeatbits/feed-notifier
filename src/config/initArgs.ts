export const initArgs = [
  '-initialize',
  '--initialize',
];
