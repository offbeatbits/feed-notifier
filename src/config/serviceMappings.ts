import {
  TwitterService,
  MastodonService,
} from '../services';

export const serviceMappings = {
  twitter: TwitterService,
  mastodon: MastodonService,
};
