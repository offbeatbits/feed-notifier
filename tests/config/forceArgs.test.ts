import { forceArgs } from "../../src/config/forceArgs";

describe("Force args", () => {
  it("should export array of force arguments", () => {
    expect(forceArgs).toMatchSnapshot();
  });
});
