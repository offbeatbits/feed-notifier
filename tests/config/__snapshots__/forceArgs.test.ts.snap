// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Force args should export array of force arguments 1`] = `
Array [
  "-skipCache",
  "--skipCache",
  "-skipcache",
  "--skipcache",
]
`;
